tests/subsys/usb_c/mocks/mock_tester.c

```diff
@@ -0,0 +1,393 @@
+
+	tester_transmit_data(&tester.tx_msg);
+}
+
+/**
+ * @breif Get tester's PD revision
+ *
+ * @retval PD revision
+ */
+enum pd_rev_type tester_get_rev(void)
+{
+	return tester.rev;
+}
+
+/**
+ * @brief Set tester's PD revision to 2
+ */
+void tester_set_rev_pd2(void)
+{
+	tester.rev = PD_REV20;
+}
+
+/**
+ * @brief Set tester's PD revision to 3
+ */
+void tester_set_rev_pd3(void)
+{
+	tester.rev = PD_REV30;
+}
+
+/**
+ * @brief Set tester's power role to Source
+ */
+void tester_set_power_role_source(void)
+{
+	tester.power_role = TC_ROLE_SOURCE;
+}
+
+/**
+ * @brief Set tester's data role to UFP
+ */
+void tester_set_data_role_ufp(void)
+{
+	tester.data_role = TC_ROLE_UFP;
+}
+
+/**
+ * @brief Set tester's data role to DFP
+ */
+void tester_set_data_role_dfp(void)
+{
+	tester.data_role = TC_ROLE_DFP;
+}
+
+/**
+ * @breif Give tester access to the TCPC's data
+ *
+ * @param tcpc pointer to TCPC's data
+ */
+void tester_set_tcpc_device(struct mock_tcpc_data *tcpc)
+{
+	tester.tcpc = tcpc;
+}
+
+/**
+ * @brief Tester performs a disconnect
+ */
+void tester_disconnected(void)
+{
+	tester_apply_vbus(0);
+	tester_apply_cc(TC_CC_VOLT_OPEN, TC_CC_VOLT_OPEN);
+	tester_msgid_set(0);
+}
+
+/**
+ * @brief Tester sets the CC lines to a voltage level
+ *
+ * @param cc1 Voltage to place on cc1
+ * @param cc2 Voltage to place on cc2
+ */
+void tester_apply_cc(enum tc_cc_voltage_state cc1, enum tc_cc_voltage_state cc2)
+{
+	tester.cc1 = cc1;
+	tester.cc2 = cc2;
+}
+
+/**
+ * @brief saves the pull resistor set by the UUT
+ *
+ * @param pull UUT's cc pull resistor
+ */
+void tester_set_uut_cc(enum tc_cc_pull pull)
+{
+	tester.uut_cc_pull = pull;
+}
+
+/**
+ * @breif Tester sets the CC lines RP value
+ *
+ * @param cc1 RP value to place on cc1
+ * @param cc2 RP value to place on cc2
+ */
+void tester_apply_rp(enum tc_rp_value cc1, enum tc_rp_value cc2)
+{
+	switch (cc1) {
+	case TC_RP_USB:
+		tester.cc1 = TC_CC_VOLT_RP_DEF;
+		break;
+	case TC_RP_1A5:
+		tester.cc1 = TC_CC_VOLT_RP_1A5;
+		break;
+	case TC_RP_3A0:
+		tester.cc1 = TC_CC_VOLT_RP_3A0;
+		break;
+	case TC_RP_RESERVED:
+		tester.cc1 = TC_CC_VOLT_OPEN;
+		break;
+	}
+
+	switch (cc2) {
+	case TC_RP_USB:
+		tester.cc2 = TC_CC_VOLT_RP_DEF;
+		break;
+	case TC_RP_1A5:
+		tester.cc2 = TC_CC_VOLT_RP_1A5;
+		break;
+	case TC_RP_3A0:
+		tester.cc2 = TC_CC_VOLT_RP_3A0;
+		break;
+	case TC_RP_RESERVED:
+		tester.cc2 = TC_CC_VOLT_OPEN;
+		break;
+	}
+}
+
+/**
+ * @brief Get CC1's voltage level
+ *
+ * @retval CC1 voltage level
+ */
+enum tc_cc_voltage_state tester_get_cc1(void)
+{
+	if (tester.power_role == TC_ROLE_SOURCE) {
+		if (tester.uut_cc_pull == TC_CC_RP || tester.uut_cc_pull == TC_CC_OPEN) {
+			return TC_CC_VOLT_OPEN;
+		}
+	} else {
+		if (tester.uut_cc_pull != TC_CC_RP) {
+			return TC_CC_VOLT_OPEN;
+		}
+	}
+
+	return tester.cc1;
+}
+
+/**
+ * @brief Get CC2's voltage level
+ *
+ * @retval CC2 voltage level
+ */
+enum tc_cc_voltage_state tester_get_cc2(void)
```

Comment:
Nit: This function has the same logic as the previous function. You could have one helper function that just takes a CC value for CC1 or CC2 as a parameter and have these two functions be one-liners defined in terms of that function.